src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -235,7 +242,19 @@ private void initSearchModifierButtons() {
         fulltextButton.setTooltip(new Tooltip(Localization.lang("Fulltext search")));
         initSearchModifierButton(fulltextButton);
         fulltextButton.selectedProperty().addListener((obs, oldVal, newVal) -> {
-            searchPreferences.setSearchFlag(SearchFlags.FULLTEXT, newVal);
+            if (!filePreferences.shouldFulltextIndexLinkedFiles() && newVal) {
+                boolean enableFulltextSearch = dialogService.showConfirmationDialogAndWait(Localization.lang("Fulltext search"), Localization.lang("Fulltext search requires the setting 'Automatically index all linked files for fulltext search' to be enabled. Do you want to enable indexing now?"), Localization.lang("Enable indexing"), Localization.lang("Keep disabled"));
+
+                LibraryTab libraryTab = tabContainer.getCurrentLibraryTab();
+                if (libraryTab != null && enableFulltextSearch) {
+                    filePreferences.setFulltextIndexLinkedFiles(true);
+                    libraryTab.getLuceneManager().rebuildIndex();
```

Comment:
No need for this, LuceneManager already has a listener for preferences. https://github.com/JabRef/jabref/blob/5ccf3e0b43fa9612a3c17f09465a23aaef3fe1fe/src/main/java/org/jabref/logic/search/LuceneManager.java#L63-L75

Reply:
ah okay, so I just need to set the value in the prefs